File: frontend/src/features/analytics/types/analytics.ts
export interface AtrasadosPorCohorte {
  fecha: string;
  cohorte: string;
  total_atrasados: number;
  total_alumnos: number;
  porcentaje: number;
}

export interface DistribucionNotas {
  rango: string;
  cantidad: number;
}

export interface PrediccionAbandono {
  alumno_id: string;
  alumno_nombre: string;
  materia: string;
  promedio: number;
  atrasos: number;
  riesgo: 'bajo' | 'medio' | 'alto';
}

export interface DashboardKpi {
  total_alumnos: number;
  total_atrasados_actual: number;
  promedio_general: number;
  alumnos_en_riesgo: { bajo: number; medio: number; alto: number };
  tendencia_atrasados_ultimo_mes: Array<{ fecha: string; total: number }>;
  total_materias: number;
}

export interface AnalyticsDashboardFilters {
  fecha_desde?: string;
  fecha_hasta?: string;
  carrera_id?: string;
  cohorte_id?: string;
  materia_id?: string;
  dictado_id?: string;
  riesgo?: 'bajo' | 'medio' | 'alto';
}

File: backend/app/schemas/analytics.py
"""Pydantic schemas for analytics module."""

from uuid import UUID
from pydantic import BaseModel, ConfigDict, Field


class AtrasadosPorCohorteItem(BaseModel):
    model_config = ConfigDict(extra="forbid")
    fecha: str  # ISO date string YYYY-MM-DD
    cohorte: str
    total_atrasados: int = Field(..., ge=0)
    total_alumnos: int = Field(..., ge=0)
    porcentaje: float = Field(..., ge=0.0, le=100.0)


class DistribucionNotasItem(BaseModel):
    model_config = ConfigDict(extra="forbid")
    rango: str  # "0-25%", "26-50%", etc.
    cantidad: int = Field(..., ge=0)


class PrediccionAbandonoItem(BaseModel):
    model_config = ConfigDict(extra="forbid")
    alumno_id: UUID
    alumno_nombre: str
    materia: str
    promedio: float = Field(..., ge=0.0)
    atrasos: int = Field(..., ge=0)
    riesgo: str  # "bajo" | "medio" | "alto"


class DashboardResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")
    total_alumnos: int = Field(..., ge=0)
    total_atrasados_actual: int = Field(..., ge=0)
    promedio_general: float = Field(..., ge=0.0)
    alumnos_en_riesgo: dict[str, int]  # {"bajo": N, "medio": N, "alto": N}
    tendencia_atrasados_ultimo_mes: list[dict]  # [{fecha, total}]
    total_materias: int = Field(..., ge=0)
